common/txmgr/address_state.go

```
@@ -117,14 +117,39 @@ func NewAddressState[
 	return &as, nil
 }
 
-// CountTransactionsByState returns the number of transactions that are in the given state
+// CountTransactionsByState returns the number of transactions that are in the given state.
 func (as *AddressState[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) CountTransactionsByState(txState txmgrtypes.TxState) int {
-	return 0
+	as.RLock()
+	defer as.RUnlock()
+
+	switch txState {
+	case TxUnstarted:
+		return as.unstartedTxs.Len()
+	case TxInProgress:
+		if as.inprogressTx != nil {
+			return 1
+		}
+		return 0
+	case TxUnconfirmed:
+		return len(as.unconfirmedTxs)
+	case TxConfirmedMissingReceipt:
+		return len(as.confirmedMissingReceiptTxs)
+	case TxConfirmed:
+		return len(as.confirmedTxs)
+	case TxFatalError:
+		return len(as.fatalErroredTxs)
+	}
+
+	return -1
```

Is it better to panic if we encounter unknown state here?